src/PointwiseFunctions/AnalyticSolutions/GrMhd/KomissarovShock.cpp

```diff
@@ -0,0 +1,284 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/GrMhd/KomissarovShock.hpp"
+
+#include <cstddef>
+#include <pup.h>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+#include "PointwiseFunctions/Hydro/LorentzFactor.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace {
+template <typename DataType>
+Scalar<DataType> compute_piecewise(
+    const tnsr::I<DataType, 3, Frame::Inertial>& x,
+    const double& shock_position, const double& left_value,
+    const double& right_value) noexcept;
+
+template <>
+Scalar<double> compute_piecewise(const tnsr::I<double, 3, Frame::Inertial>& x,
+                                 const double& shock_position,
+                                 const double& left_value,
+                                 const double& right_value) noexcept {
```

Comment:
Agreed, and often you lose something (performance) because the compiler ends up treating it as a pointer under the hood and you have more instructions (not many, but still :) )

Reply:
Right, this was left over from when I tried to have a generic implementation for the scalar and vector function. Fixed.